File: BubbleSort.java
import java.util.ArrayList;
import java.util.Collections;

public class BubbleSort {
	public static ArrayList<Integer> bsort(ArrayList<Integer> dataList) {
		for(int index = 0; index < dataList.size() - 1; index++) {
			boolean swap = false;
			
			for(int index2 = 0; index2 < dataList.size() - 1; index2++) {
				if(dataList.get(index2) > dataList.get(index2 + 1)) {
					Collections.swap(dataList, index2, index2 + 1);
					swap = true;
				}
			}
			
			if(swap == false) {
				break;
			}
		}
		
		return dataList;
	}
	
	public static void main(String[] args) {
		ArrayList<Integer> testData = new ArrayList<Integer>();
		ArrayList<Integer> resultData = new ArrayList<Integer>();
		
		for (int i = 0; i < 100; i++) {
			testData.add((int)(Math.random() * 100));
		}
		
		System.out.println(testData);
		
		resultData = bsort(testData);
		
		System.out.println(resultData);
	}
}

File: Sort/BubbleSort.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
using namespace std;

void bsort(vector<int> arr, int size) {
	for (int index = 0; index < size; index++) {
		for (int index2 = 0; index2 < size - index - 1; index2++) {
			if (arr[index2] > arr[index2 + 1]) {
				int temp = arr[index2];
				arr[index2] = arr[index2 + 1];
				arr[index2 + 1] = temp;
			}
		}
	}

	for (int i = 0; i < size; i++) {
		cout << arr[i] << " ";
	}
	
}

int main() {
	vector<int> testData(10);

	for (int i = 0; i < 10; i++) {
		testData[i] = int(rand() % 100);
	}
	
	
	for (int i = 0; i < 10; i++) {
		cout << testData[i] << " ";
	}
	cout << endl;

	bsort(testData, testData.size());

	return 0;
}
